ionc/ion_writer.c

```diff
@@ -3281,3 +3282,14 @@ iERR _ion_writer_free_temp_pool( ION_WRITER *pwriter )
 
     iRETURN;
 }
+
+iERR _ion_writer_free_pending_pool(ION_WRITER *writer) {
+    iENTER;
+
+    if (writer->_pending_temp_entity_pool != NULL) {
+        ion_free_owner(writer->_pending_temp_entity_pool);
+        writer->_pending_temp_entity_pool = NULL;
+    }
+
+    RETURN(__location_name__, __line__, __count__++, err);
```

Comment:
Is this an intentional choice not to use `iRETURN`?

Should this just be a `void` function since it looks like it can't return anything but `IERR_OK`?

Reply:
Yea that was intentional. Since the `fail` label that `iRETURN` defines isn't used, using it over `RETURN` only adds to the warnings generated at build.

It could definitely be a `void` function. I maintained the signature that the other free functions had for writer members, but all of them could be converted to void since all 3 only return `IERR_OK`.

Reply:
Got it. So maybe we need a separate (and larger) task to get rid of unnecessary `iRETURN` usages, if we want to reduce warnings. Approving this PR.